ui/src/test/java/edu/wpi/grip/ui/PaletteTest.java

```diff
@@ -0,0 +1,70 @@
+package edu.wpi.grip.ui;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.util.Modules;
+import edu.wpi.grip.core.AdditionOperation;
+import edu.wpi.grip.core.Operation;
+import edu.wpi.grip.core.Step;
+import edu.wpi.grip.core.events.OperationAddedEvent;
+import edu.wpi.grip.core.events.StepAddedEvent;
+import edu.wpi.grip.util.GRIPCoreTestModule;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+import org.junit.Test;
+import org.testfx.framework.junit.ApplicationTest;
+import org.testfx.util.WaitForAsyncUtils;
+
+import javax.inject.Inject;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class PaletteTest extends ApplicationTest {
+
+    @Inject private EventBus eventBus;
+
+    private Step step;
+    private final GRIPCoreTestModule testModule = new GRIPCoreTestModule();
+
+    @Override
+    public void start(Stage stage) throws Exception {
+        testModule.setUp();
+
+        Injector injector = Guice.createInjector(Modules.override(testModule).with(new GRIPUIModule()));
+        injector.injectMembers(this);
+
+        FXMLLoader loader = new FXMLLoader(getClass().getResource("Palette.fxml"));
+        loader.setControllerFactory(injector::getInstance);
+        stage.setScene(new Scene(loader.load(), 300, 800));
+        stage.show();
+    }
+
+    @Override
+    public void stop() {
+        testModule.tearDown();
+    }
+
+    @Test
+    public void testPalette() {
+        // Given a single operation...
+        Operation addition = new AdditionOperation();
+        eventBus.post(new OperationAddedEvent(addition));
+
+        // If we click on the operation button in the palette...
```

Comment:
I would create an anonymous inner class here instead that did the subscribe.
You should avoid test variables from escaping the system under test.
If other tests are added `step` may not be null because of another test.

The same instance of this test class is reused between every test so you should isolate the output for the test to the scope of the method.